bin/data-get.js

```diff
@@ -69,27 +69,30 @@ const run = async () => {
       // Try to guess owner and dataset name here. We're not loading Dataset object
       // because we want to handle private datasets as well:
       const idParts = identifier.split('/')
-      const owner = idParts[idParts.length - 2]
-      const name = idParts[idParts.length - 1]
-      const token = config.get('token')
-      pathToSave = path.join(owner, name)
-
-      if (!checkDestIsEmpty(owner, name)) {
-        throw new Error(`${owner}/${name} is not empty!`)
+      if (idParts.length === 5) {
+        const owner = idParts[idParts.length - 2]
+        const name = idParts[idParts.length - 1]
+        const token = config.get('token')
+        pathToSave = path.join(owner, name)
+
+        if (!checkDestIsEmpty(owner, name)) {
+          throw new Error(`${owner}/${name} is not empty!`)
+        }
+        
+        /** For datasets from the datahub we get zipped version and unzip it.
+                - less traffic
+                - zipped version has a fancy file structure
+            #issue: https://github.com/datahq/datahub-qa/issues/86  */
+        const zipped_dataset_url  = `https://datahub.io/${owner}/${name}/r/${name}_zip.zip?jwt=${token}`
+        const archive_path = await saveFileFromUrl(zipped_dataset_url, 'zip')
+        // unzip archive into destination folder
+        fs.createReadStream(archive_path)
+          .pipe(unzip.Extract({ path: pathToSave }))
+          // removing the archive file once we extracted all the dataset files
+          .on('finish', () => {fs.unlinkSync(archive_path)})
+      } else {
+        pathToSave = await saveFileFromUrl(identifier, argv.format)
```

Comment:
It's really hard to follow the logic here. Does this mean it will download everything that has the identifier length not equal to 5? if so why it's not done after line 96 where it says `// If it is not a dataset - download the file`

To be honest `if (idParts.length === 5) {...}` smalls very bedly think will lead to bunch of other bugs

Reply:
I think logic should be:

```javascript
if (identifier.includes('/r/') {
  pathToSave = await saveFileFromUrl(identifier, argv.format)
} else {
  // previous logic
}
```

so we assume that if `/r/` is in identifier then it's `r` link.